### DISCUSSION TITLE
Container -> child context enforcement

### DISCUSSION THREAD

**Discussion by yinonov** [Vonage/vivid#858] [Q&A]
enforcement of context and property value is often required where components are part of an integration within a collective.
in such cases it is advisable to enable a single source of appliance to such permutation rather than on each child element.
allowing individual value to each child within a collective can result in a "circus" like UI.
list & its list-items (as in #833), button-toggle-group, textfield and its action buttons (as explained [here](https://github.com/Vonage/vivid/pull/837#issuecomment-846516656))
there are 2 methods to apply such enforcement

- css variables
- programmatic meddling on child components

in the recent [PR on interactive](#837) icons I experienced both practices trying to consolidate 1 of the 2 practices as a rule of thumb.
BUT
thinking on each various use-cases separately leads to different choice and I feel we auto-decide on the best approach for each use-case

**Comment by lagjamhero** [comment]
I agree with handling collective elements with props and css from the top level. 
Reduces repetition and chance of mistakes for the consumer

**Reply by yinonov** [reply]
see [related comment](https://github.com/Vonage/vivid/pull/833#issuecomment-846979857)
